### ISSUE TITLE
OpenAPIv3: Code-assist: Custom `title` properties in JSON Schema

### ISSUE THREAD

**Issue by tfesenko** [RepreZen/KaiZen-OpenAPI-Editor#296]
Extracted from #291. 
We added several custom `title` properties to the JSON Schema for Swagger v2 in https://github.com/RepreZen/KaiZen-OpenAPI-Editor/pull/108/commits/3b7a12026a77f997b86a5765cd3899d3e43ddfcb 
Check if we need custom `title`s in OpenAPI v3 and update the Schema generator(#292) accordingly.

**Comment by tedepstein** [comment]
Adding to current scope, pending clarification: Will our 3.0 content assist and/or validation use these by default, or will we need to (re-)add custom logic for this purpose?

**Comment by tfesenko** [comment]
We probably won't need custom `title` properties. The code which calculates human-friendly label does use `title` if it's available, but the value of `$ref` is good enough. 
* This code is already used by validation
* It's used by prototype for code-assist for #227 

This logic is currently implemented in [MultipleSwaggerErrorBuilder.getHumanFriendlyText(JsonNode, String)](https://github.com/RepreZen/KaiZen-OpenAPI-Editor/blob/master/com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/validation/MultipleSwaggerErrorBuilder.java#L108), but it will be moved to a more appropriate location when we decided to use it for code-assist too.

I don't see any cases where we need the custom `title`s. Closing this issue for now. We can always reopen it if needed.
